File: packages/core/src/feedback/FeedbackLoop.ts
import { FeedbackEvent, FeedbackEventType, createFeedbackEvent } from "./FeedbackEvent";

export class FeedbackLoop {
  private events: FeedbackEvent[];
  private sessionId: string;

  constructor(sessionId: string) {
    this.sessionId = sessionId;
    this.events = [];
  }

  recordEvent(
    type: FeedbackEventType,
    data?: Record<string, unknown>
  ): FeedbackEvent {
    const event = createFeedbackEvent(type, this.sessionId, data);
    this.events.push(event);
    return event;
  }

  getEvents(): FeedbackEvent[] {
    return [...this.events];
  }

  getEventsByType(type: FeedbackEventType): FeedbackEvent[] {
    return this.events.filter((event) => event.type === type);
  }

  clearEvents(): void {
    this.events = [];
  }
}

export function createFeedbackLoop(sessionId: string): FeedbackLoop {
  return new FeedbackLoop(sessionId);
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/feedback/FeedbackLoop.kt
package ai.mobileagentos.compose.feedback

class FeedbackLoop(private val sessionId: String) {
    private val events: MutableList<FeedbackEvent> = mutableListOf()
    
    fun recordEvent(type: FeedbackEventType, data: Map<String, Any>? = null): FeedbackEvent {
        val event = FeedbackEvent.create(type, sessionId, data)
        events.add(event)
        return event
    }
    
    fun getEvents(): List<FeedbackEvent> = events.toList()
    
    fun getEventsByType(type: FeedbackEventType): List<FeedbackEvent> {
        return events.filter { it.type == type }
    }
    
    fun clearEvents() {
        events.clear()
    }
}